test/smartwallet/CustomSmartWallet.test.ts

```diff
@@ -0,0 +1,1197 @@
+                assert.isFalse(await smartWallet.isInitialized());
+                //Initializing the contract
+                await assert.isRejected(
+                    smartWallet.initialize(
+                        senderAddress,
+                        customLogic.address,
+                        null,
+                        worker,
+                        '0',
+                        '400000',
+                        '0x'
+                    )
+                );
+
+                assert.isFalse(await smartWallet.isInitialized());
+            });
+
+            it('Should verify method initialize reverts with negative gas amount', async () => {
+                //Making sure the contract has not been initialized yet
+                assert.isFalse(await smartWallet.isInitialized());
+                //Initializing the contract
+                await assert.isRejected(
+                    smartWallet.initialize(
+                        senderAddress,
+                        customLogic.address,
+                        token.address,
+                        worker,
+                        '100',
+                        '-400000',
+                        '0x'
+                    )
+                );
+
+                assert.isFalse(await smartWallet.isInitialized());
+            });
+
+            it('Should verify method initialize reverts with negative token amount', async () => {
+                //Making sure the contract has not been initialized yet
+                assert.isFalse(await smartWallet.isInitialized());
+                //Initializing the contract
+                await assert.isRejected(
+                    smartWallet.initialize(
+                        senderAddress,
+                        customLogic.address,
+                        token.address,
+                        worker,
+                        '-2',
+                        '400000',
+                        '0x'
+                    )
+                );
+
+                assert.isFalse(await smartWallet.isInitialized());
+            });
+
+            it('Should verify method initialize successfully with token as 0x address parameter', async () => {
+                //Making sure the contract has not been initialized yet
+                assert.isFalse(await smartWallet.isInitialized());
+                //Initializing the contract
+                await assert.isFulfilled(
+                    smartWallet.initialize(
+                        senderAddress,
+                        customLogic.address,
+                        constants.ZERO_ADDRESS,
+                        worker,
+                        '10',
+                        '400000',
+                        '0x'
+                    )
+                );
+
+                assert.isTrue(await smartWallet.isInitialized());
+            });
+
+            it('Should verify method initialize successfully with worker address as null parameter', async () => {
```

Comment:
Is this test description correct? The assert says isFalse.

Reply:
Good catch, thank you for highlight this. In my opinion, I think that this test could be removed, because we set the address with a null value where the null value shouldn't be allowed; indeed it's an error for typescript, but it looks like truffle doesn't perform the ts check before running the tests. Maybe we could try to configure truffle to check the types when running the tests.

Reply:
For now, I've updated the description to reflect what it actually does